js/digifinex.js

```diff
@@ -2261,10 +2313,21 @@ module.exports = class digifinex extends Exchange {
         const query = this.omit (params, this.extractParams (path));
         const urlencoded = this.urlencode (this.keysort (query));
         if (signed) {
-            const nonce = this.nonce ().toString ();
-            const auth = urlencoded;
-            // the signature is not time-limited :\
-            const signature = this.hmac (this.encode (auth), this.encode (this.secret));
+            let signature = undefined;
+            let nonce = undefined;
+            if (pathPart === '/swap/v2') {
+                nonce = this.milliseconds ().toString ();
+                let preHash = nonce + method + payload;
+                if (urlencoded) {
+                    preHash += '?' + urlencoded;
+                }
+                signature = this.hmac (this.encode (preHash), this.encode (this.secret));
```

Comment:
we're repeating this code twice: `signature = this.hmac (this.encode (preHash), this.encode (this.secret));` , can we dry it?

Reply:
Yeah, I just added an updated version of sign